### ISSUE TITLE
consider setting cache-control:immutable

### ISSUE THREAD

**Issue by kralo** [cdnjs/cdnjs#8655]
which is a [cache-control extension](https://tools.ietf.org/html/rfc7234#section-5.2.3) implemented already in [firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1267474). I think this is reasonable, as you already set a large cache expiration date `Cache-Control: public, max-age=30672000`

This could help reducing calls to the server resulting in 304 not-modified and thereby also reduce some privacy issues (less requests - less traceability)...

**Comment by drewfreyling** [comment]
This is a good idea but might not be of real benefit until it is supported by more browsers as it is just a bunch of wasted bytes unless you use a recent version of Firefox.
Chrome doesn't seem to be in a hurry - https://bugs.chromium.org/p/chromium/issues/detail?id=611416

**Comment by PeterDaveHello** [comment]
cc @cdnjs/owners @terinjokes

**Comment by terinjokes** [comment]
This looks pretty cool.

I don't think we should implement it until I change the backend to reduce the update failures we have from time to time (and, perhaps, officially supporting "latest")

**Comment by dima74** [comment]
Hello! What about adding `Cache-Control: immutable` only when user-agent is Firefox?

**Comment by PeterDaveHello** [comment]
Hi @terinjokes, I think it'll be great to enable it no matter it the backend on Cloudflare side changes as this can really help improve performance by reducing the cache validation, am I right?

**Comment by MattIPv4** [comment]
@PeterDaveHello Chrome still does not have this implemented, should I raise this to Cf to ask them to set Cache control to be immutable when a Firefox UA is detected?

**Comment by MatthewSteeples** [comment]
[Everyone apart from Chrome (and probable IE)](https://stackoverflow.com/questions/41936772/which-browsers-support-cache-control-immutable) implement this now

**Comment by MattIPv4** [comment]
@MatthewSteeples Yeah, I think it best we implement it as it should improve performance. I will contact Cf and post any updates here :)

**Comment by MattIPv4** [comment]
@xtuc would you be able to help with getting this added?

**Comment by xtuc** [comment]
According to https://caniuse.com/#feat=mdn-http_headers_cache-control_immutable the support is pretty low

**Comment by MattIPv4** [comment]
Currently, we send `cache-control: public, max-age=30672000`. Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Caching_static_assets it should be safe to append `, immutable`.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Extension_Cache-Control_directives notes that if a UA doesn't recognize `immutable`, it should just be ignored, so it'd fall back to the long max-age we are sending.

(I'm no expert on this though)

**Comment by xtuc** [comment]
@MattIPv4 that sound right. I sent the header to my Chrome and it didn't fail. That said, I'm not sure how to check that it works

**Comment by MattIPv4** [comment]
@xtuc Simple. Just wait a year and see if it expires. /s

**Comment by MattIPv4** [comment]
Chrome's dev tools should show if the resource was fetched from cache or not, but I'm not sure of any way to really confirm how long it's planning to hold the resource for in cache.
